### ISSUE TITLE
Get string representation of partial HTML selection

### ISSUE THREAD

**Issue by slybridges** [timdown/rangy#412]
Hi Tim & all - first of all, many thanks for rangy!

I have a use case where I would need to get the exact string of a partial HTML selection and/or range.

Example, let's suppose i make the following selection (represented by the part within `*`)
`<div>Some people *say this <b>Goat* cheese</b> is the finest of all.</div>`

In this scenario:
- `toString()` would return `say this Goat`
- `toHtml()` would return `say this <b>Goat</b>` (<-- browser has fixed malformed HTML)

What I would like to get the string `say this <b>Goat`

Is that at all possible with Rangy ? If not, would anybody have a suggestion on how implement such method ?

Cheers!

**Comment by slybridges** [comment]
Following up on this old issue.

What I ended up doing at first was to play with full nodes innerHTML properties along with the node and offsets given by rangy and recreate the partial selection... but I dropped the approach altogether in the end as browsers can't guarantee that innerHTML will return exactly the same code source as the one you gave initially (that was a prerequisite for me).

Especially Firefox seems to reorder element attributes in an alphabetical order. Fun!

In a nutshell: if you need to preserve your HTML source code but use the DOM API from the browser  to manipulate the DOM and read it back, you're doing it wrong :-) Try to push things server-side where at least you can control that everything will be dealt with in the same way.

Closing as there is nothing much rangy can do here.
